### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by thisHermit**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+  * [Reporting Bugs](#reporting-bugs)
+  * [Suggesting Enhancements](#suggesting-enhancements)
+  * [Your First Code Contribution](#your-first-code-contribution)
+  * [Pull Requests](#pull-requests)
+
+[Styleguides](#styleguides)
+  * [Git Commit Messages](#git-commit-messages)
+  * [JavaScript Styleguide](#javascript-styleguide)
+  * [CoffeeScript Styleguide](#coffeescript-styleguide)
+  * [Specs Styleguide](#specs-styleguide)
```
Comment (amjaps) [inline CONTRIBUTING.md]: Shouldn't we use Dart Styleguide?

**Review by amjaps — CHANGES_REQUESTED** [review]
Everything is fine. There are some lines mentioning about javascript, CoffeeScript, and Python files

File: CONTRIBUTING.md (hunk 1 above)
Comment (thisHermit) [reply CONTRIBUTING.md]: Sure, will fix that.

**Review by thisHermit — COMMENTED** [review]
Sure will do that

**Review by amjaps — APPROVED** [review]

**Pull request merged**